### ISSUE TITLE
Support for tests

### ISSUE THREAD

**Issue by Razzeee** [microsoft/language-server-protocol#1267]
With testing explorer coming to vscode soonish, it might be nice to discuss if this can be included in the protocol too. Apparently they choose how they wanted to implement this, but decided against something like lsp (or even making it part of lsp?)

https://code.visualstudio.com/updates/v1_56#_testing

**Comment by dbaeumer** [comment]
We are discussing how we can open up LSP to non language requests which would ease to add something like a test provider.

**Comment by mickaelistria** [comment]
This seems related to https://github.com/Microsoft/vscode-debugadapter-node/issues/154 .
It would be great to provide a Test Adapter Protocol model that both LSP and DAP could share.

**Comment by mfussenegger** [comment]
The [Build Server Protocol](https://build-server-protocol.github.io/) also covers parts of running tests or tasks. Maybe worth having a look if LSP is to be extended with such functionality.

**Comment by davidanthoff** [comment]
The primary thing that I'd like to see is protocol support so that a LS can notify a client about the location and existence of tests. I don't think test execution needs to be part of the LSP.

**Comment by firelizzard18** [comment]
The Go language server (gopls) is planning on adding a [custom command for test discovery](https://github.com/golang/go/issues/59445).

**Comment by davidanthoff** [comment]
Yes, that is what we did in the Julia LS as well, add a custom notification that reports test locations back to the client.

**Comment by dbaeumer** [comment]
A first version of the base protocol is available here that should ease defining new protocols like test: https://microsoft.github.io/language-server-protocol/specifications/base/0.9/specification/

@MariaSolOs
